src/v/redpanda/monitor_unsafe_log_flag.cc

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "redpanda/monitor_unsafe_log_flag.h"
+
+#include "cluster/logger.h"
+#include "utils/utf8.h"
+
+using cluster::clusterlog;
+
+class monitor_unsafe_log_flag::handle_unsafe_log_flag_change {
+public:
+    explicit handle_unsafe_log_flag_change(
+      cluster::cluster_version original_version)
+      : _original_version(original_version)
+      , _legacy_permit_unsafe_log_operation(
+          config::shard_local_cfg().legacy_permit_unsafe_log_operation.bind()) {
+        handle_flag_change();
+        _legacy_permit_unsafe_log_operation.watch(
+          [this] { handle_flag_change(); });
+    }
+
+    ss::future<> stop() { co_return; }
```

Comment:
I do not think you **need** to defined a stop method. The `ss::sharded<T>::stop` is defined in SS already and it will call optionally call `T::stop()` if it exists.